ecommerce/orders/serializers.py

```diff
@@ -0,0 +1,147 @@
+from django.db.transaction import atomic
+from rest_framework import serializers
+
+from orders.models import (BillingAddress, Order, OrderBankAccount,
+                           OrderItem, ShippingAddress)
+from customers.serializers import CustomerSerializer, CitySerializer
+from baskets.serializers import BasketSerializer
+
+
+class BillingAddressSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = BillingAddress
+        read_only_fields = ['id', 'created_at', 'updated_at']
+        fields = '__all__'
+
+
+class ShippingAddressSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ShippingAddress
+        read_only_fields = ['id', 'created_at', 'updated_at']
+        fields = '__all__'
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Order
+        read_only_fields = ['id', 'created_at', 'updated_at']
+        fields = '__all__'
+
+
+class OrderItemSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = OrderItem
+        read_only_fields = ['id', 'created_at', 'updated_at']
+        fields = '__all__'
+
+
+class OrderBankAccountSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = OrderBankAccount
+        read_only_fields = ['id', 'created_at', 'updated_at']
+        fields = '__all__'
+
+
+class OrderDetailedSerializer(OrderSerializer):
+    customer = CustomerSerializer()
+    basket = BasketSerializer()
+    billing_address = BillingAddressSerializer()
+    shipping_address = ShippingAddressSerializer()
+
+    @atomic()
+    def create(self, validated_data, order=None):
```

Comment:
detailed serializerlari sadece veri gosterimi icin kullaniyoruz, ozel durumlar yoksa nested serializerlari create/update icin kullanmiyoruz, ayrica bu serializer ozelinde konusacak olursak customer, address vs bu asamada olusmus olmali

Reply:
Anladım hocam product detailed serializer'da oluşturduğumuz için bende hepsinde oluşturmaya çalıştım birbiri ile foreign key ilişkisi olan tablolar arası. Ama kaçırdığım şey yukarda belirttiğiniz gibi basket oluştururken neden customer oluşturalım ? :) Böylece sizin product detailed serializer da yaptığınızla farkı daha anladım